File: Java/groupAnagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> ans= new ArrayList<>();
        HashMap<String, List<String>> check= new HashMap<>();

        for(String s: strs){
            char[] charArr= s.toCharArray();
            Arrays.sort(charArr);

            String temp= String.valueOf(charArr);
            if(!check.containsKey(temp))
                check.put(temp, new ArrayList<>());
            
            check.get(temp).add(s);
        }

        ans.addAll(check.values());
        return ans;
    }
}

File: Python/groupAnagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        check = defaultdict(list)

        for s in strs:
            temp = ''.join(sorted(s))
            check[temp].append(s)
        
        return list(check.values())
